From 8d446bb8047193bdb607fd47276768d096bdec68 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Mon, 10 Feb 2025 10:31:24 +0100
Subject: [PATCH 1531/1695] plat-stm32mp2: remove unnecessary check from
 pwr_set_voltage()

"res" variable has already been checked previously.
Remove dead code from pwr_set_voltage().
Legacy code introduced when regulator_get_voltage() used to return an
error code.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 06190d8e0da0e68d250fba8091c0ea15db4ebd4f plat-stm32mp2: remove unnecessary check from pwr_set_voltage()]
Change-Id: I4b80102421a816b0f04fa7bc445814c9d0a8381f
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/435810
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c
index 3ea5e85dc..c26003a0c 100644
--- a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c
+++ b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c
@@ -294,8 +294,6 @@ static TEE_Result pwr_set_voltage(struct regulator *regulator, int level_uv)
 		     regulator_name(regulator), result);
 		/* Continue to restore IOs setting for current voltage */
 		level_uv = regulator_get_voltage(regulator->supply);
-		if (res)
-			return res;
 	}
 
 	if (level_uv < IO_VOLTAGE_THRESHOLD_UV) {
-- 
2.47.2

